src/handlers/updateUserProfile.ts

```diff
@@ -20,14 +20,19 @@ export const schema = {
 
 export async function handler(req: ValidatedRequest<typeof schema>, res: express.Response) {
   if (!req.params.id) {
-    res.status(400).send()
+    res.status(400).send('No User ID was provided.')
     return
   }
-  if (req.user?.d_id !== req.params.id) {
-    res.status(403).send()
+  if (!req.user?.d_id) {
+    res.status(401).send('You need to be logged in.')
```

Comment:
The original code was meant to ensure you could only update your own user.

As now this endpoint should allow either:
- any authenticated user to update his own profile: `req.params.id === req.user?.d_id`
- any authenticated admin to update any profile: `req.user?.flags === 1` regardless of which user id is being updated.

Those conditions can be checked here, instead the process done in lines 30-34. By doing that we skip a database query.